guides/custom_interpretations_with_blocks.md

```diff
@@ -0,0 +1,186 @@
+<script type="module" src="https://gradio.s3-us-west-2.amazonaws.com/3.0.24/gradio.js"></script>
+
+# Custom Machine Learning Interpretations with Blocks
+Docs: Blocks
+Tags: INTERPRETATION, SENTIMENT ANALYSIS
+
+**Prerequisite**: This Guide requires you to know about Blocks and the interpretation feature of Interfaces.
+Make sure to [read the Guide to Blocks first](/introduction_to_blocks) as well as the
+interpretation section of the [Advanced Interface Features Guide](/advanced_interface_features).
+
+## Introduction
+
+If you have experience working with the Interface class, then you know that interpreting the prediction of your machine learning model
+is as easy as setting the `interpretation` parameter to either "default" or "shap".
+
+You may be wondering if it is possible to add the same interpretation functionality to an app built with the Blocks API.
+Not only is it possible, but the flexibility of Blocks lets you display the interpretation output in ways that are
+impossible to do with Interfaces!
+
+This guide will show how to:
+
+1. Recreate the behavior of Interfaces's interpretation feature in a Blocks app.
+2. Customize how interpretations are displayed in a Blocks app.
+
+Let's get started!
+
+## Setting up the Blocks app
+
+Let's build a sentiment classification app with the Blocks API.
+This app will take text as input and output the probability that this text expresses either negative or positive sentiment.
+We'll have a single input `Textbox` and a single output `Label` component.
+Below is the code for the app as well as the app itself.
+
+```python
+import gradio as gr 
+from transformers import pipeline
+
+sentiment_classifier = pipeline("text-classification", return_all_scores=True)
+
+def classifier(text):
+    pred = sentiment_classifier(text)
+    return {p["label"]: p["score"] for p in pred[0]}
+
+with gr.Blocks() as demo:
+    with gr.Row():
+        with gr.Column():
+            input_text = gr.Textbox(label="Input Text")
+            with gr.Row():
+                classify = gr.Button("Classify Sentiment")
+        with gr.Column():
+            label = gr.Label(label="Predicted Sentiment")
+
+    classify.click(classifier, input_text, label)
+demo.launch()
+```
+
+<gradio-app space="freddyaboulton/sentiment-classification"> </gradio-app>
```

Comment:
Hosting on spaces so that we don't have to download model weights on the website right now.